internal/fs/vss_windows.go

```diff
@@ -716,7 +727,7 @@ func HasSufficientPrivilegesForVSS() bool {
 		return false
 	}
 
-	return !(HRESULT(result) == E_ACCESSDENIED)
+	return HRESULT(result) != E_ACCESSDENIED
```

Comment:
This code unreachable if `result` == `E_ACCESSDENIED` because `err` always != `nil` if `hresult` != `S_OK`.
`initializeVssCOMInterface` should not return `HRESULT`. If you still need it in other places you may use something like this:
```
if e, ok := err.(*vssError); ok {
    // Use e.hresult
}
```
And if you want your custom message from `initializeVssCOMInterface` to be seen by user, `HasSufficientPrivilegesForVSS` should return `error`, but not `bool`.

Reply:
Fixed in https://github.com/restic/restic/pull/3045/commits/947f00e9ec9237ae07d04072721d2c67fce7adf5